### PR TITLE
Specific interface for provider enhancement (3/3)

### PR DESCRIPTION
**Pull request by colerogers**
Change for the the specific crashlytics interface. Exposes `onNewFatalIssuePublished`, `onNewNonfatalIssuePublished`, `onRegressionAlertPublished`, `onStabilityDigestPublished`, `onVelocityAlertPublished`, and `onNewAnrIssuePublished` from the `v2/alerts/crashlytics` namespace

Ex ~ 
```
import {
  onNewFatalIssuePublished,
  onNewNonfatalIssuePublished,
  onRegressionAlertPublished,
  onStabilityDigestPublished,
  onVelocityAlertPublished,
  onNewAnrIssuePublished
} from "firebase-functions/v2/alerts/crashlytics";

// newFatalIssuePublished interface
export const func = onNewFatalIssuePublished((event) => { … });
export const func1 = onNewFatalIssuePublished("123456789", (event) => { … });
export const func2 = onNewFatalIssuePublished({ appId: "123456789" }, (event) => { … });
// newNonfatalIssuePublished interface
export const func3 = onNewNonfatalIssuePublished((event) => { … });
export const func4 = onNewNonfatalIssuePublished("123456789", (event) => { … });
export const func5 = onNewNonfatalIssuePublished({ appId: "123456789" }, (event) => { … });
// regressionAlertPublished interface
export const func6 = onRegressionAlertPublished((event) => { … });
export const func7 = onRegressionAlertPublished("123456789", (event) => { … });
export const func8 = onRegressionAlertPublished({ appId: "123456789" }, (event) => { … });
// stabilityDigestPublished interface
export const func9 = onStabilityDigestPublished((event) => { … });
export const func10 = onStabilityDigestPublished("123456789", (event) => { … });
export const func11 = onStabilityDigestPublished({ appId: "123456789" }, (event) => { … });
// velocityAlertPublished interface
export const func12 = onVelocityAlertPublished((event) => { … });
export const func13 = onVelocityAlertPublished("123456789", (event) => { … });
export const func14 = onVelocityAlertPublished({ appId: "123456789" }, (event) => { … });
// newAnrIssuePublished interface
export const func15 = onNewAnrIssuePublished((event) => { … });
export const func16 = onNewAnrIssuePublished("123456789", (event) => { … });
export const func17 = onNewAnrIssuePublished({ appId: "123456789" }, (event) => { … });
```

### CODE REVIEW

File: src/v2/providers/alerts/crashlytics.ts [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+
+/** Data */
+/** Generic issue interface */
+interface Issue {
+  id: string;
+  title: string;
+  subtitle: string;
+  appVersion: string;
+}
+/** Payload for a new fatal issue */
```
Comment (inlined) [inline src/v2/providers/alerts/crashlytics.ts:14]: nit: newlines between struct definitions

File: src/v2/providers/alerts/crashlytics.ts [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+export interface NewNonfatalIssuePayload {
+  ['@type']: 'com.google.firebase.firebasealerts.CrashlyticsNewNonfatalIssuePayload';
+  issue: Issue;
+}
+/** Payload for a regression alert */
+export interface RegressionAlertPayload {
+  ['@type']: 'com.google.firebase.firebasealerts.CrashlyticsRegressionAlertPayload';
+  type: string;
+  issue: Issue;
+  resolveTime: string;
```
Comment (inlined) [inline src/v2/providers/alerts/crashlytics.ts:29]: FYI: I don't recall all the details, but we have an outstanding bug for the auth namespace that the proto-JSON format isn't the SDK format for date strings. Let's not make the same mistake again.

Comment (inlined) [reply src/v2/providers/alerts/crashlytics.ts:29]: https://github.com/firebase/firebase-functions/issues/1012 for details. Let's bring this up with API council and see if there needs to be an AIP on the standard way going forwards.

File: src/v2/providers/alerts/crashlytics.ts [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+export interface VelocityAlertPayload {
+  ['@type']: 'com.google.firebase.firebasealerts.VelocityAlertPayload';
+  issue: Issue;
+  createTime: string;
+  crashCount: number;
+  crashPercentage: number;
+  firstVersion: string;
+}
+/** Payload for a new ANR issue */
+export interface NewAnrIssuePayload {
```
Comment (inlined) [inline src/v2/providers/alerts/crashlytics.ts:54]: Nit: I wonder if we should spell out Application Not Responding in the docstring

File: src/v2/providers/alerts/crashlytics.ts [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+    ) => any | Promise<any>;
+    appIdOrOptsOrHandler = {};
+  }
+
+  const [opts, appId] = getOptsAndApp(
+    appIdOrOptsOrHandler as string | CrashlyticsOptions
+  );
+
+  const func = (raw: CloudEvent<unknown>) => {
+    return handler(raw as CrashlyticsEvent<T>);
```
Comment (inlined) [inline src/v2/providers/alerts/crashlytics.ts:295]: Here's where you might need to recognize some date fields and change their format (if API council decides that ISO 8601 isn't the right format). I know you've already gone through review, but it just might be worth bringing up to a few people (e.g. @mbleigh) that we have inconsistencies and need an AIP.

**Review by inlined — APPROVED** [review]
Holy moly 🥵

That's a thorough API and test suite with a heck of a terse/efficient implementation. Excellent job!

File: src/v2/providers/alerts/crashlytics.ts (hunk 1 above)
Comment (colerogers) [reply src/v2/providers/alerts/crashlytics.ts]: fixed

File: src/v2/providers/alerts/crashlytics.ts (hunk 3 above)
Comment (colerogers) [reply src/v2/providers/alerts/crashlytics.ts:77]: changed and added to other doc strings to make them more descriptive

File: src/v2/providers/alerts/crashlytics.ts [hunk 5]
```diff
@@ -69,42 +104,38 @@ export const velocityAlert = 'crashlytics.velocity';
...
-/** Cloud Event Type */
-interface WithAlertTypeAndApp {
-  alertType: string;
-  appId: string;
-}
-export type CrashlyticsEvent<T> = CloudEvent<
-  FirebaseAlertData<T>,
-  WithAlertTypeAndApp
->;
+/** @internal */
```
Comment (inlined) [inline src/v2/providers/alerts/crashlytics.ts:114]: Please go back to inlining the callback type. It improves code complete in VS Code.

File: src/v2/providers/alerts/index.ts [hunk 6]
```diff
@@ -1,5 +1 @@
-import * as crashlytics from './crashlytics';
-
-export { crashlytics };
```
Comment (inlined) [inline src/v2/providers/alerts/index.ts]: why did we yank this?

**Review by inlined — APPROVED** [review]

File: src/v2/providers/alerts/crashlytics.ts (hunk 5 above)
Comment (colerogers) [reply src/v2/providers/alerts/crashlytics.ts]: Removed the handler type in the latest commit

File: src/v2/providers/alerts/index.ts (hunk 6 above)
Comment (colerogers) [reply src/v2/providers/alerts/index.ts]: Added the import/export back in the latest commit

**Pull request merged**